driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -521,7 +521,7 @@ void onDown(Host host) throws InterruptedException, ExecutionException {
     // not necessary
     // since updateCreatedPools should take care of it. But better protect against non well behaving
     // policies.
-    removePool(host).force().get();
+    removePool(host).get();
```

Comment:
Wait for inflight requests to complete before closing connections. If they don't end up being closed they should be shutdown automatically by `ConnectionReaper` after 2x`readTimeoutMillis`